### ISSUE TITLE
USB DAC crashes when phone is locked

### ISSUE THREAD

**Issue by F1VWdk** [GrapheneOS/os-issue-tracker#1631]
Observed on Pixel 7 Pro and 6
Pixel 7 Pro Version: [TD1A.220804.031.2022102800]
No additional drivers/software installed for the DAC
Spotify plays Ok while phone is unlock or locked when output to Bluetooth headset
with USB DAC:
while phone is unlocked, connect USB DAC (BTR5 by FiiO)
start Spotify, it plays ok to USB DAC while phone is unlocked
the second phone is locked, sound stops. Unlocking phone causes Spotify to crash without restoring sound. Once started again, Spotify plays fine to the DAC while phone is unlocked, locking phone repeats the cycle.
Observed similar behavior using Amazon Music but unlike Spotify, AM did not crash after unlocking the phone but instead reported "exception 800" and needed to be forced closed and restarted to resume play.

This playback configuration was working on the the same phone with Android 13 (and 12)

**Comment by wonder75** [comment]
Maybe try the ANKER DAC. It works without problems for me on Pixel 6a.
Had problems with another DAC too (iMagoo).

**Comment by psz1234** [comment]
> Observed on Pixel 7 Pro and 6 Pixel 7 Pro Version: [TD1A.220804.031.2022102800] No additional drivers/software installed for the DAC Spotify plays Ok while phone is unlock or locked when output to Bluetooth headset with USB DAC: while phone is unlocked, connect USB DAC (BTR5 by FiiO) start Spotify, it plays ok to USB DAC while phone is unlocked the second phone is locked, sound stops. Unlocking phone causes Spotify to crash without restoring sound. Once started again, Spotify plays fine to the DAC while phone is unlocked, locking phone repeats the cycle. Observed similar behavior using Amazon Music but unlike Spotify, AM did not crash after unlocking the phone but instead reported "exception 800" and needed to be forced closed and restarted to resume play.
> 
> This playback configuration was working on the the same phone with Android 13 (and 12)

This is a BUG in GOS. All other ROMS that I tried do not have this issue. I've tried Google Stock ROM, LineageOS, DivestOS, CalxOS. None of these have this problem.

**Comment by thestinger** [comment]
Need to know if anyone can replicate the issue on a Pixel 4 or Pixel 5.

**Comment by thestinger** [comment]
@psz1234 An issue being uncovered on GrapheneOS usually doesn't mean it's a bug in GrapheneOS but rather a bug in AOSP uncovered by memory corruption detection features. Stop posting comments adding nothing to the discussion if you want to participate here.

**Comment by b3K1ndRw1nd** [comment]
> Need to know if anyone can replicate the issue on a Pixel 4 or Pixel 5.

@thestinger I can confirm that the issue does not happen (at least the music stopping when phone is locked) on my Pixel 4 that is also running grapheneOS.

Is there anything I can do to help resolve this issue?

**Comment by maade69** [comment]
it seems like I can't replicate this issue on Pixel 7 with Apple Type-C to 3.5 dongle

**Comment by psz1234** [comment]
This is a Pixel 6 problem with GOS. Only people with Pixel 6 are experiencing this issue as far as I can tell.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by thestinger** [comment]
The initial post says that it's occurring for them with a Pixel 7 Pro. Maybe it only happens with certain kinds of DACs.

**Comment by b3K1ndRw1nd** [comment]
@maade69 will respond to [this comment](https://github.com/GrapheneOS/os-issue-tracker/issues/1643#issuecomment-1320191871) here to centralize all discussion about this issue

[these](https://www.amazon.ca/gp/product/B081294BZB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) are the earphones I have. Nothing special.

**Comment by SHA-NDY** [comment]
Same for me with Pixel 6. I used my USB-C audio converter (https://www.jsaux.com/products/jsaux-usb-c-to-3-5mm-cable-adapter) successfully back in September, that's all I can say on when it worked the last time for me (don't really use it often). Maybe I was still on Android 12 back then.

**Comment by empratyush** [comment]
@F1VWdk  i am unable to reproduce the issue with [BTR5 on pixel 7](https://drive.google.com/file/d/1BRDCXzBMIa2gVeEfiXv1HqlUCSD9LNrX/view?usp=share_link).

**Comment by empratyush** [comment]
@shape-err  I am unable to reproduce the with[ Google USB-C to 3.5 mm Adaptor ](https://store.google.com/product/usb_c_headphone_adapter) and Pixel 6 pro.

How i can reproduce this bug (step by step) on what build number (settings > about > build number)?

**Comment by psz1234** [comment]
I am not allowed to post

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by empratyush** [comment]
> @empratyush Build Number: TP1A.221105.002.2022111800 Steps to Reproduce:
> 
> 1. Plug in USB/3.5 Adapter (Phone can either be locked or unlocked)
> 2. Connect Headphones/Audio device to adapter
> 3. Open app to play audio (Spotify, Soundcloud, YouTube, etc)
> 4. Play audio (While phone is unlocked audio will play)
> 5. Lock device by power button
> 6. Audio will immediately stop, fingerprint reader freezes for a few seconds
> 7. Unlock device, the app that was playing audio and any app that can play audio will crash/freeze upon opening until adapter is unplugged

I am sorry but still unable to reproduce this issue .

**Comment by b3K1ndRw1nd** [comment]
@empratyush 

for me, I have a Google Pixel 6 with Build Number 221105.002.2022111800 and my steps to reproduce is:

1. install https://github.com/AdrienPoupa/VinylMusicPlayer
2. connect https://www.amazon.ca/gp/product/B081294BZB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
3. start playing song
4. lock Phone
5. after a second or so, the music will stop playing

To get the song to start playing again, I have to
1. Unlock
2. disconnect earphones
3. press pause
4. connect earphones
5. press play

**Comment by b3K1ndRw1nd** [comment]
I can confirm that @shape-err 's "fix" does work on my Pixel 6 too.

**Comment by SHA-NDY** [comment]
Confirmed. Works for me too, Pixel 6.
Thanks guys for investigating.

**Comment by b3K1ndRw1nd** [comment]
@empratyush wait, is this expected behavior? cause this was not how the phones used to work.

**Comment by SHA-NDY** [comment]
@empratyush I always had "Data transfer" as default setting for USB. It used to work before as mentioned by @b3K1ndRw1nd

**Comment by SHA-NDY** [comment]
@psz1234 Yes, Pixel 6 (not Pro)

**Comment by empratyush** [comment]
> @empratyush wait, is this expected behavior? cause this was not how the phones used to work.

yes, by default "Default USB Configuration " is set to "No data transfer" for privacy/security reasons.

**Comment by thestinger** [comment]
I don't think this issue is resolved. I think people are messing with developer options in a way that they shouldn't be. OS level file transfer enabling is not the same as that.

**Comment by SHA-NDY** [comment]
> You shouldn't need to change anything in developer options. The developer options USB transfer mode is a different thing than the user facing one. Does this work for you with developer options disabled completely?

For me it works as expected with developer options disabled completely (Pixel 6)

**Comment by b3K1ndRw1nd** [comment]
> I don't think this issue is resolved. I think people are messing with developer options in a way that they shouldn't be. OS level file transfer enabling is not the same as that.

@thestinger Not sure if I am misunderstanding you or you are legit saying that users should not be enabling USB file sharing on their pixels?

**Comment by SHA-NDY** [comment]
> > I don't think this issue is resolved. I think people are messing with developer options in a way that they shouldn't be. OS level file transfer enabling is not the same as that.
> 
> @thestinger Not sure if I am misunderstanding you or you are legit saying that users should not be enabling USB file sharing on their pixels?

I think changing the default behaviour of USB by setting developer options is messing this up.
But I always had developer options enabled like that and it wasn't a problem regarding music playback.

**Comment by thestinger** [comment]
If everyone here previously changed the developer options setting for this, then that's the issue. Developer options are for testing and aren't intended to be used this way.

**Comment by SHA-NDY** [comment]
> If everyone here previously changed the developer options setting for this, then that's the issue. Developer options are for testing and aren't intended to be used this way.

Accepted

**Comment by b3K1ndRw1nd** [comment]
> If everyone here previously changed the developer options setting for this, then that's the issue. Developer options are for testing and aren't intended to be used this way.

@thestinger is there another option for doing a USB file transfer to and from a Pixel phone?

**Comment by thestinger** [comment]
It's in the standard USB menu in Settings -> Connected devices. There's no reason to use developer options for this and it's broken. When you connect a USB device, a notification is shown with a shortcut to the USB settings menu in Settings.

**Comment by b3K1ndRw1nd** [comment]
I see. well I weekly use the `adb` command so its gonna suck to keep having to toggle it on and off so that the earphones can work now :disappointed: but thanks I guess

**Comment by thestinger** [comment]
You don't have to do that. ADB doesn't cause any issues with this.

**Comment by thestinger** [comment]
You're wrong. It works the same way on the stock OS based on the same version of AOSP. It's also not an opinion that developer options are intended for developers and not production use. There are a whole bunch of broken ones, ones with major unexplained caveats and ones that don't do anything. Features are in developer options because they're only useful for developers, extremely dangerous (ADB) or too half baked / broken to be exposed as regular options for users. Please don't unnecessarily bump issues and do research before trying to contradict things stated by the GrapheneOS team. You could confirm for yourself this occurs on stock OS.

**Comment by thestinger** [comment]
> Without changing Developer options, every time you connect a phone to a computer, you have to slide down and select the operation mode.

Aside from not being implemented properly in a way that doesn't break things, I think the reason they haven't done that and added it as a normal option is fairly obvious: even using the regular approach that needs to be enabled each time is a quite insecure approach to file transfer where the attached computer gets access to the user's whole home directory. Having that happen automatically without the user unlocking and enabling it is a pretty big issue...
